### PR TITLE
Add digest source 'tag' to use tag without digest

### PR DESCRIPTION
**Pull request by fengye87**
**Related**: #3649 

**Description**

A new option `skaffold render --digest-source=tag` was added, which will render the manifests with built tags. The difference between 'none' and 'tag' is that the former one will keep whatever artifacts are in the original manifests, while the latter will update them with proper tags, without sha256 digests.

This could be very helpful when rendering one-for-all release manifests in multi-arch deployment scenarios, although extra cares should be taken in this case to not move any released tags.

**User facing changes (remove if N/A)**

User gets a new option `skaffold render --digest-source=tag`

### CODE REVIEW

**Review by tejal29 — APPROVED** [review]

File: pkg/skaffold/runner/build_deploy.go [hunk 1]
```diff
@@ -50,7 +50,8 @@ func (r *SkaffoldRunner) Build(ctx context.Context, out io.Writer, artifacts []*
 	}
 
 	// In dry-run mode or with --digest-source  set to 'remote', we don't build anything, just return the tag for each artifact.
```
Comment (MarlonGamez) [inline pkg/skaffold/runner/build_deploy.go:52]:
```suggestion
	// In dry-run mode or with --digest-source  set to 'remote' or 'tag', we don't build anything, just return the tag for each artifact.
```

**Review by MarlonGamez — CHANGES_REQUESTED** [review]
This may be a dumb question but, what is the difference between using 'remote' vs 'tag'? Since they both go down the same code path now

**Review by MarlonGamez — APPROVED** [review]
LGTM, just one nit to update the comment

**Comment by fengye87** [comment]
@tejal29 I've updated the comment as @MarlonGamez suggested, could you start another kokoro run please?

**Pull request merged**